nav2_collision_monitor/src/polygon.cpp

```diff
@@ -129,6 +148,15 @@ void Polygon::getPolygon(std::vector<Point> & poly) const
   poly = poly_;
 }
 
+bool Polygon::isShapeSet()
+{
+  if (poly_.empty()) {
+    RCLCPP_WARN(logger_, "[%s]: Polygon shape is not set yet", polygon_name_.c_str());
```

Comment:
Make this a throttled warning, since this would blast the entire terminal at a high rate until set